addons/hr_attendance/models/hr_attendance.py

```diff
@@ -393,21 +392,20 @@ def _update_overtime(self, employee_attendance_dates=None):
                         affected_employees |= overtime.employee_id
                 elif overtime:
                     overtime_to_unlink |= overtime
+        # Create new overtime records
         created_overtimes = self.env['hr.attendance.overtime'].sudo().create(overtime_vals_list)
+        # Collect all affected employees
         employees_worked_hours_to_compute = (affected_employees.ids +
-                                             created_overtimes.employee_id.ids +
-                                             overtime_to_unlink.employee_id.ids)
+                                            created_overtimes.employee_id.ids +
+                                            overtime_to_unlink.employee_id.ids)
         overtime_to_unlink.sudo().unlink()
-        to_recompute = self.search([('employee_id', 'in', employees_worked_hours_to_compute)])
-        # for automatically validated attendances, avoid recomputing extra hours if user has changed its value
-        validated_modified = to_recompute.filtered(lambda att: att.employee_id.company_id.attendance_overtime_validation == 'no_validation'
-                                                        and float_compare(att.overtime_hours, att.validated_overtime_hours, precision_digits=2))
-        self.env.add_to_compute(self._fields['overtime_hours'],
-                                to_recompute)
-        self.env.add_to_compute(self._fields['validated_overtime_hours'],
-                                to_recompute - validated_modified)
-        self.env.add_to_compute(self._fields['expected_hours'],
-                                to_recompute)
+        to_recompute = self.search([('employee_id', 'in', employees_worked_hours_to_compute),
+                                    ('check_out', '!=', False),
+                                    ('check_in', '>=', datetime.combine(min(all_dates, default=datetime(1000, 1, 1)), datetime.min.time())),
+                                    ('check_in', '<=', datetime.combine(max(all_dates, default=datetime(3000, 1, 1)), datetime.max.time()))])
+        self.env.add_to_compute(self._fields['overtime_hours'], to_recompute)
+        self.env.add_to_compute(self._fields['validated_overtime_hours'], to_recompute)
+        self.env.add_to_compute(self._fields['expected_hours'], to_recompute)
```

Comment:
unnecessary diff. In stable, the diff should be minimal to fix the issue, no need to refactor/reformat the code if it is not needed. 🙂